### PR TITLE
xtest: 1011: test RPC via secure memory too

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Extends test case 1011 to test RPC both via normal shared memory and
secure shared memory.

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

File: host/xtest/xtest_1000.c [hunk 1]
```diff
@@ -749,16 +749,32 @@ static void xtest_tee_test_1011(ADBG_Case_t *c)
 		TA_RPC_CMD_CRYPT_AES256ECB_ENC,
 		TA_RPC_CMD_CRYPT_AES256ECB_DEC
 	};
+	struct xtest_crypto_session cs_secshm = {
+		c, &session, TA_RPC_CMD_CRYPT_SHA256,
+		TA_RPC_CMD_CRYPT_AES256ECB_ENC,
+		TA_RPC_CMD_CRYPT_AES256ECB_DEC
+	};
```
Comment (lackan) [inline host/xtest/xtest_1000.c]: Here should be s/TA_RPC_CMD_CRYPT/TA_RPC_CMD_CRYPT_SECSHM ?

Comment (jenswi-linaro) [inline host/xtest/xtest_1000.c]: Thanks, yes you are right.

**Comment by jenswi-linaro** [comment]
Update

**Comment by etienne-lms** [comment]
@jenswi-linaro, I fell uncomfortable with the references to "secure shared memory" and "RPC" especially used together.

xtest 1011 is more about "TA-to-TA session test" rather than "RPC test" despite legacy test related labels. And your change is more about "TA private memory" rather than specific "secure shared memory".

If you agree, few labeling and the commit comment would neede to be updated, i.e:
```
xtest: 1011: extend to test TA-to-TA and TA private memory references

Extends test case 1011 to test TA-to-TA session both via normal shared memory
reference parameters and TA private memory reference parameters.
```

Aside from this and @lackan comment (i saw you just pushed a fix), change is ok to me.

File: host/xtest/xtest_1000.c [hunk 2]
```diff
@@ -751,8 +751,8 @@ static void xtest_tee_test_1011(ADBG_Case_t *c)
 	};
 	struct xtest_crypto_session cs_secshm = {
 		c, &session, TA_RPC_CMD_CRYPT_SHA256,
```
Comment (lackan) [inline host/xtest/xtest_1000.c]: Hi, Jens, TA_RPC_CMD_CRYPT_SHA256 should also change?

**Comment by lackan** [comment]
Without the comment, Test-by: Guanchao Liang <liang.guanchao@linaro.org> (FVP)

File: host/xtest/xtest_1000.c (hunk 2 above)
Comment (jenswi-linaro) [inline host/xtest/xtest_1000.c]: Yes, you're right.

**Comment by jenswi-linaro** [comment]
Updated

**Comment by lackan** [comment]
Reviewed-by: Guanchao Liang <liang.guanchao@linaro.org>
Tested-by: Guanchao Liang <liang.guanchao@linaro.org> (FVP)

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@linaro.org>`

Extra comments (could be part of another change):
- `TA_RPC_CMD_CRYPT_PRIVMEM_SHA224` and `TA_RPC_CMD_CRYPT_SHA224` are not used. Should be removed or used ?
- It could be nice to test also TA-to-staticTA with bot ns-shm and TA privmem reference params.

**Comment by jenswi-linaro** [comment]
Tags applied

**Comment by jenswi-linaro** [comment]
Rebased.

**Comment by jforissier** [comment]
Once more please ;)

**Comment by jenswi-linaro** [comment]
Rebased :-)

**Pull request merged**